wkwebview/PersonaViewController.swift

```diff
@@ -8,30 +8,49 @@
 import UIKit
 import WebKit
 
+/// Delegate methods that get called once a verification is completed.
+protocol PersonaViewControllerDelegate: class {
+    /// Verification completed successfully.
+    func verificationSucceeded(viewController: PersonaViewController, inquiryId: String)
+
+    /// Verification failed.
+    func verificationFailed(viewController: PersonaViewController)
+}
+
+/// Wrapper around a web view that handles the verification.
 class PersonaViewController: UIViewController {
 
-    lazy private var webView: WKWebView = {
-        let webView = WKWebView()
-        webView.navigationDelegate = self
-        webView.allowsBackForwardNavigationGestures = false
-        webView.frame = view.frame
-        webView.scrollView.bounces = false
-        return webView
-    }()
+    /// The URL to redirect to once the verification is complete
+    private let redirectUri = "https://personademo.com"
+
+    // The web view.
+    private let webView = WKWebView()
+
+    // The delegate that gets called when verification is complete.
+    weak var delegate: PersonaViewControllerDelegate?
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        addWebView()
 
-        // Add the web view
-        view.addSubview(webView)
+        // The dictionary with the Persona configuration options
+        // See the Persona docs (http://documentation.withpersona.com) for full documentation.
+        let config = [
+            "template-id": "tmpl_JAZjHuAT738Q63BdgCuEJQre",
+            "environment": "sandbox",
+            "reference-id": "myReference",
+            "redirect-uri": redirectUri,
+            "is-webview": "true"
+        ]
 
-        // Load the Persona URL request
-        let request = createPersonaInitializationURLRequest()
-        webView.load(request)
-    }
+        // Create query items for the configuration dictionary
+        var components = URLComponents(string: "https://withpersona.com/verify")
+        components?.queryItems = config.map { URLQueryItem(name: $0, value: $1) }
```

Comment:
Could you inline the `config` query parameters directly instead of making the config object and then mapping it to the appropriate class?

Reply:
@SalvatoreT completely agree. i found the dictionary easier to read, but mapping it later is a little confusing. I've pushed that change.